### PR TITLE
Fix AHBot Buyer to stop counting own auctions

### LINKED ISSUE #15542: [335] AHBot Buyer Considers Own Auctions In Chance Calcs.
__Core:__ 747ea7ca2c2b757dc9c5a672746d801a8365ee98
__DB:__ TDB_full_335.59_2015_07_14 + updates

__Bug:__

When calculating the *chance*, the Buyer uses *ahInfo->BuyItemCount* to determine if it needs to calculate a weighted chance. Unfortunately it considers its own auctions in this number which it can never buy.

The effect is the possibility of a very small buyout chance when player and bot sell the same item.

__Details:__

I sold 1x Small Brown Pouch (entry 4496) but the bot has 44 of them for sale. When this line was evaluated:

```C++
// If there are more than 5 items on AH of this entry, try weigh in the average buyout price
if (ahInfo->BuyItemCount > 5)
    chance *= 1.f / std::sqrt(itemBuyPrice / avgBuyPrice);
```
my auction was given a very low chance of buyout (__8.86255455__) because the bot thinks I'm selling 45 of them, even though I have the config set to a very high chance:

> AuctionHouseBot.Buyer.ChanceFactor = 90
> AuctionHouseBot.Buyer.ChanceMultiplier.Gray = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.White = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.Green = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.Blue = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.Purple = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.Orange = 1000
> AuctionHouseBot.Buyer.ChanceMultiplier.Yellow = 1000

In debug, that same auction when skipping that calculation ended up with a __27.2220325__ chance which seems more accurate since I'm only selling one and my buyout (1g 25s) is 100 times the sell price (1s 25c).

Since the bot can never buy its own auctions, it should not be considering them in formulas regarding bid or buyout of player auctions.

__Notes:__

The chance calculations use *BuyerItemInfo::ahInfo* but *BuyerItemInfo* does not contain the seller's information so there is no way for the calculations to know who's auction it is.

### PR DESCRIPTION
**Pull request by MrSmite**
closes #15542

### CODE REVIEW

**Comment by MrSmite** [comment]
Hmm, bot still needs to look at bids which this prevents.

**Comment by MrSmite** [comment]
Note: The comment at the top of the function says:

```C++
// Collects information about item counts and minimum prices to SameItemInfo and updates EligibleItems - a list with new items eligible for bot to buy and bid
// Returns count of items in AH that were eligible for being bought or bidded on by ahbot buyer (EligibleItems size)
```

Since the bod can't buy / bid on its own auctions, there is no reason to consider any bot auction for *EligibleItems* either

**Comment by MrSmite** [comment]
Hmm, can't seem to squash the commits. I keep getting "everything up to date" as a result.

**Comment by jackpoz** [comment]
it doesn't even build

File: src/server/game/AuctionHouseBot/AuctionHouseBotBuyer.cpp [hunk 1]
```diff
@@ -135,11 +139,10 @@ uint32 AuctionBotBuyer::GetItemInformation(BuyerConfiguration& config)
                 itemInfo.MinBuyPrice = std::min(itemInfo.MinBuyPrice, itemBuyPrice);
         }
 
-        // Add/update to EligibleItems if:
-        // has a bid by player or
-        // has no bids and not owned by bot
-        if ((entry->bid && entry->bidder) || (entry->owner && !entry->bid))
-        {
```
Comment (Shauren) [inline src/server/game/AuctionHouseBot/AuctionHouseBotBuyer.cpp]: where did the bracket go?

**Comment by tkrokli** [comment]
You still haven't fixed your git e-mail address, I see.

File: src/server/game/AuctionHouseBot/AuctionHouseBotBuyer.cpp (hunk 1 above)
Comment (MrSmite) [inline src/server/game/AuctionHouseBot/AuctionHouseBotBuyer.cpp]: > it doesn't even build
> where did the bracket go?

Sorry, I edited it directly on github. Must've hit delete too many times.

Comment (MrSmite) [inline src/server/game/AuctionHouseBot/AuctionHouseBotBuyer.cpp]: > where did the bracket go?

That doesn't even match my source file

```
if (!entry->bid || entry->bidder)
```

But you're right, I did accidentally remove a bracket. I was editing directly in github :(

**Comment by MrSmite** [comment]
> You still haven't fixed your git e-mail address, I see. (github doesn't recognize the owner of your commits.)

Curious, why does it matter re: recognizing the owner?

Anyway, I did forget about this, should be fixed now.

**Comment by MrSmite** [comment]
> @jackpoz it doesn't even build
> @Shauren  where did the bracket go?

I was editing directly in github :(

**Comment by tkrokli** [comment]
Yes, I see you have fixed the git e-mail address now. Well done.

**Comment by jackpoz** [comment]
GitHub Online Editor is evil

**Comment by MrSmite** [comment]
> GitHub Online Editor is evil

Yeah, I had an extra line after the brace so I quickly edited it but I must've hit too many backspaces.

*I blame Firefox for input lag ;)*

**Comment by Shauren** [comment]
>Yeah, I had an extra line after the brace so I quickly edited it but I must've hit too many backspaces.
>I blame Firefox for input lag ;)

Thats not the most evil part - it allows you to create files that have names starting with space

**Comment by Aokromes** [comment]
Well, you can do it on linux too :p
(bad escuse from github guis to allow it....)

**Comment by DDuarte** [comment]
You can do these mistakes in a lot of ways, the problem is that the online editor makes it too easy to do them.

**Comment by MrSmite** [comment]
Thanks for the merge.

Curious: I have a "Revert" button next to the message that Shauren merged the commit. Is this due to the linking of my GIT e-mail that @tkrokli was asking me to do? Can't say I've ever seen that button before.

**Comment by DDuarte** [comment]
That button creates a PR to revert this PR. Probably a new thing on Github (before it only appeared to committers).

**Pull request merged**